### DISCUSSION TITLE
How to get an IReactiveObject from a templated control?

### DISCUSSION THREAD

**Discussion by Gomez12** [AvaloniaUI/Avalonia#11689] [Q&A]
I am completely new with Avalonia and MVVM.

But I am running into a problem.

Which is that I can't see a way to retrieve the ReactiveObject from a templated control.

What I currently have is a lot of code in the viewmodel which works.
But I would like to split it up to separate (templated) controls to maintain readability etc.

But for example I can't seem to get the following code to work in the code-behind of the templated control : 
```c#
private string? _searchText;
public string? SearchText
{
    get => _searchText;
    set => this.RaiseAndSetIfChanged(ref _searchText, value);
}
```
Because this is a TemplatedControl and not a ViewModelBase / ReactiveObject.

Am I missing something? Should I use user-controls instead of templated controls?

**Comment by timunie** [comment]
Please check out Avalonia.Samples. We've a sample about custom controls there.
